static/js/controllers/contacts.js

```diff
@@ -219,6 +224,63 @@ var _ = require('underscore'),
         $scope.search();
       };
 
+      $scope.simprintsEnabled = !!(window.medicmobile_android &&
+          window.medicmobile_android.simprints_available &&
+          window.medicmobile_android.simprints_available());
+      $scope.triggerSimprintsIdent = function() {
+        /* jslint bitwise: true */
+        var simprintsInputId = Math.floor(Math.random() * Number.MAX_SAFE_INTEGER) & SP_ID_MASK;
+        console.log('Launching simprints with input ID: ' + simprintsInputId);
+
+        var $input = $($element).find('input[name=simprints-idents]');
+        $input.attr('data-simprints-idents', simprintsInputId);
+        $input.on('change', function() {
+          var $this = $(this);
+          $this.off('change');
+          handleSimprintsIdentResponse(JSON.parse($this.val()));
+        });
+
+        $scope.loading = true;
+        window.medicmobile_android.simprints_ident(simprintsInputId);
+      };
+
+      var handleSimprintsIdentResponse = function(idents) {
+        console.log('handleSimprintsIdentResponse()', idents);
+
+        $scope.loading = true;
+        $scope.filtered = true;
+        liveList = LiveList['contact-simprints-search'];
+        liveList.set([]);
+
+        DB().query('medic-client/people_by_simprints_id', {
+          keys: _.pluck(idents, 'id'),
+          include_docs: true,
+        })
+          .then(function(results) {
+            var docs = _.pluck(results.rows, 'doc');
+            docs.forEach(function(doc) {
+              var ident = idents.find(function(ident) {
+                return ident.id === doc.simprints_id;
+              });
+              doc._simprints_confidence = ident.confidence;
```

Comment:
Do not modify the doc. If this doc is ever persisted this will end up in the DB and be replicated and shown to everyone! This is where the ViewModel pattern could be useful.

Also, this is purely a UI issue, but I feel like showing stars is too much info - wouldn't it be better just to order them by how closely they match?